pandas/core/series.py

```diff
@@ -5553,6 +5554,78 @@ def between(
 
         return lmask & rmask
 
+    def case_when(
+        self,
+        *args: tuple[tuple],
+        level: int | None = None,
+    ) -> Series:
+        """
+        Replace values where the conditions are True.
+        It is a convenience wrapper around `pd.Series.mask`,
+        and is handy for multiple conditions and replacements.
+
+        Parameters
+        ----------
+        *args : tuple
+            Variable argument of tuples of conditions and expected replacements.
+            Takes the form:  `(condition0, replacement0)`,
+            `(condition1, replacement1)`, ... .
+            `condition` should be a 1-D boolean array-like object
+            or a callable. If `condition` is a callable,
+            it is computed on the Series
+            and should return a boolean Series or array.
+            The callable must not change the input Series
+            (though pandas doesn`t check it). `replacement` should be a
+            1-D array-like object, a scalar or a callable.
+            If `replacement` is a callable, it is computed on the Series
+            and should return a scalar or Series. The callable
+            must not change the input Series
+            (though pandas doesn`t check it).
+
+        level : int, default None
+            Alignment level if needed.
+
+            .. versionadded:: 2.2.0
+
+        Returns
+        -------
+        Series
+
+        See Also
+        --------
+        Series.mask : Replace values where the condition is True.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({
+        ...     "a": [0,0,1,2],
+        ...     "b": [0,3,4,5],
+        ...     "c": [6,7,8,9]
+        ... })
+        >>> df
+           a  b  c
+        0  0  0  6
+        1  0  3  7
+        2  1  4  8
+        3  2  5  9
+
+        >>> df.c.case_when((df.a.gt(0), df.a),  # condition, replacement
+        ...                (df.b.gt(0), df.b))
+        0    6
+        1    3
+        2    1
+        3    2
+        Name: c, dtype: int64
+        """
+        args = [
+            (
+                com.apply_if_callable(condition, self),
+                com.apply_if_callable(replacement, self),
+            )
+            for condition, replacement in args
+        ]
```

Comment:
We haven't yet validated the arguments, resulting in less-friendly error messages. Maybe separate out the validation logic to a function we can call?